apps/astarte_realm_management_api/lib/astarte_realm_management_api/triggers/action.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Astarte.RealmManagement.API.Triggers.Action do
+  alias Astarte.RealmManagement.API.Triggers.{AMQPAction, HttpAction}
+
+  @behaviour Ecto.Type
+
+  @impl true
+  def type, do: :map
+
+  @impl true
+  def cast(reward = %{"amqp_exchange" => _}) do
+    EctoMorph.cast_to_struct(reward, AMQPAction)
```

Comment:
I think instead of using ecto morph here, if you want validations you should call

```elixir
Astarte.RealmManagement.API.Triggers.AMQPAction.changeset(reward)
```